diff -ur --unidirectional-new-file testartifactid/pom.xml testartifactid.new/pom.xml
--- testartifactid/pom.xml	2021-04-19 20:19:28.582332796 +0200
+++ testartifactid.new/pom.xml	2021-04-19 20:23:36.980364513 +0200
@@ -69,6 +69,25 @@
           <artifactId>maven-project-info-reports-plugin</artifactId>
           <version>3.0.0</version>
         </plugin>
+        <plugin>
+          <groupId>org.codehaus.mojo</groupId>
+          <artifactId>build-helper-maven-plugin</artifactId>
+          <version>3.2.0</version>
+          <executions>
+            <execution>
+              <id>add-source</id>
+              <phase>generate-sources</phase>
+              <goals>
+                <goal>add-source</goal>
+              </goals>
+              <configuration>
+                <sources>
+                  <source>${project.build.directory}/something</source>
+                </sources>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
       </plugins>
     </pluginManagement>
   </build>
diff -ur --unidirectional-new-file testartifactid/src/main/java/testpackage/App.java testartifactid.new/src/main/java/testpackage/App.java
--- testartifactid/src/main/java/testpackage/App.java	2021-04-19 20:19:28.582332796 +0200
+++ testartifactid.new/src/main/java/testpackage/App.java	2021-04-19 20:26:27.115079639 +0200
@@ -1,5 +1,7 @@
 package testpackage;
 
+import anothertestpackage.GeneratedSomething;
+
 /**
  * Hello world!
  *
@@ -8,6 +10,7 @@
 {
     public static void main( String[] args )
     {
-        System.out.println( "Hello World!" );
+        GeneratedSomething gs = new GeneratedSomething();
+        System.out.println( "Hello World!" + gs.x );
     }
 }
diff -ur --unidirectional-new-file testartifactid/target/something/anothertestpackage/GeneratedSomething.java testartifactid.new/target/something/anothertestpackage/GeneratedSomething.java
--- testartifactid/target/something/anothertestpackage/GeneratedSomething.java	1970-01-01 01:00:00.000000000 +0100
+++ testartifactid.new/target/something/anothertestpackage/GeneratedSomething.java	2021-04-19 20:28:39.566104040 +0200
@@ -0,0 +1,4 @@
+package anothertestpackage;
+public class GeneratedSomething {
+  public int x;
+}
